packages/wdio-utils/src/initialiseServices.js

```diff
@@ -5,64 +5,158 @@ import initialisePlugin from './initialisePlugin'
 const log = logger('@wdio/utils:initialiseServices')
 
 /**
- * initialise services based on configuration
- * @param  {Object}    config  config of running session
- * @param  {Object}    caps    capabilities of running session
- * @param  {String}    type    define sub type of plugins (for services it could be "launcher")
- * @return {Object[]}          list of service classes that got initialised
+ * Maps list of services of a config file into a list of actionable objects
+ * @param  {Object}    config            config of running session
+ * @param  {Object}    caps              capabilities of running session
+ * @return {[(Object|Class), Object][]}  list of services with their config objects
  */
-export default function initialiseServices (config, caps, type) {
+function initialiseServices (services) {
     const initialisedServices = []
-
-    if (!Array.isArray(config.services)) {
-        return initialisedServices
-    }
-
-    for (let serviceName of config.services) {
-        let serviceConfig = {}
-
+    for (let [serviceName, serviceConfig] of services) {
```

Comment:
I don't think destructuring like this will work when the service is an object instead of an array

Reply:
Or we need to update the example below from:
```
         * services: [{
         *     beforeTest: () => { ... }
         * }]
```
to:
```
         * services: [
                [{
         *         beforeTest: () => { ... }
         *     }]
         * ]
```